src/helpers/compress.rs

```diff
@@ -0,0 +1,32 @@
+//! # Value Compression
+
+/// # Example
+/// ```
+/// use programming_team_code_rust::helpers::compress::compress;
+///
+/// let a_and_b = [-4, 3, 100, 3, 0];
+/// let (mut a, max_val) = compress(&a_and_b);
+/// let b = a.split_off(3);
+/// assert_eq!(a, [0, 2, 3]);
+/// assert_eq!(b, [2, 1]);
+/// assert_eq!(max_val, 4);
```

Comment:
this is my idea for a good example: instead of tons of function calls, make it simple, and also add a common use case: like here compressing multiple arrays